### DISCUSSION TITLE
[Q] when I load buffer,exceljs throw error: Invalid row number in model,what may make this happen?

### DISCUSSION THREAD

**Discussion by monologueH** [exceljs/exceljs#2411] [Q&A]
![image](https://user-images.githubusercontent.com/30610963/93842490-b7c52c80-fcc9-11ea-85a6-040abe48197f.png)

**Comment by nandakumar-a** [comment]
will you provide more details about what are doing with buffer ?, if you share your code then it will help

**Comment by oste** [comment]
I am seeing this issue as well. It happens when reading a two sheet excel. If I remove the second sheet it's able to read the first sheet. Unfortunately the sheet is confidential but will try to redact it to provide a test case.

**Comment by nandakumar-a** [comment]
```
workbook.eachSheet((sheet, id) =>
   {
        sheet.eachRow((row, rowIndex) =>
        {
            console.log(row.values);
         });
});
```

@oste  will you try this?

**Comment by oste** [comment]
@nandakumardtc This error occurs during .load like this

```
  const workbook = new Excel.Workbook();
  await workbook.xlsx.load(fileBuffer);
```

So there isn't an opportunity to do what you suggest.

**Comment by nandakumar-a** [comment]
@oste can you give me more information like  are you reading a excel file using FileReader() ?

**Comment by alietors** [comment]
Hi,

I'm getting the same issue, it's a bit odd, I'll do my best to explain it.
We generate xlsx file on jsreport (not exceljs responsibility), I've attached a file for reference.

Once we have the file downloaded, we need to upload it for processing (now we use exceljs) and it fails with the mentioned error.
Here is the code
```
const workbook = new Excel.Workbook()
const localFile = await workbook.xlsx.readFile(TEMP_FILE)
```

TEMP_FILE is just the path to the file on a temp folder.

The "funny" thing here is:
If you open the xlsx file with excel and just save it (no change, just open and save) the file can be processed normally
If you use jsreport ui interface, generate the file, it will open on a browser integrated excel, save a copy and then try to process it, it will work fine.

My guess is jsreport generates the file in an unorthodox way, that excel has no problem to open but exceljs struggles with.

Here is an example of a failing file:
[staged_data.xlsx](https://github.com/exceljs/exceljs/files/6464604/staged_data.xlsx)


EDIT:
Digging a bit into it, it seems that for some reason the rows don't have a number (it comes as NaN), probably when saving on excel it "fix" the issue

**Comment by viniciusls** [comment]
Hi, I'm getting the same issue here. I'm downloading a Buffer from S3 and trying to load it using exceljs. Could any of you figure out what's wrong and how to fix it?

Thanks in advance!

**Comment by PADIGAVISHNU** [comment]
Help me with this error
Invalid row number when I am trying to load a file that is downloaded.

const workbook = new ExcelJS.Workbook();
const buffer = await file.arrayBuffer();
await workbook.xlsx.load(buffer);

if I once open the file in Excel software save it and then open the file using ExcelJS it is working fine
How to fix this bug.

Thanks in advance
